Slms/issue 10

alexredle opened the pull request:
Se agregan servicio para administración de comentarios para publicaciones.

josedepaz committed:
MOD: Typos and redaction

alexredle committed:
ADD: Servicio de comments

alexredle committed:
Agrega archivo .env

alexredle committed:
MOD: Se actualiza Validator con valores de entidad

garfenter commented on comments-ms/config.json:
```diff
@@ -0,0 +1,70 @@
+    "context": {
+        "$filter": "env",
+        "$default": {
+            "basePath": "api/v1/comment"
```
El standard que hemos utilizado en los otros servicios es el recurso colocarlo en plural. i.e. "comments"

garfenter commented on comments-ms/config.json:
```diff
@@ -0,0 +1,70 @@
+            "basePath": "api/v1/comment"
+        },
+        "production": {
+            "basePath": "api/v1/comment"
```
Igual aquí.

garfenter commented on comments-ms/handlers.js:
```diff
@@ -0,0 +1,95 @@
+
+// Create a new COMMENT
+exports.createComment = function (request, reply) {
+    const COMMENT = {
```
Esta variable debería de estar en minusculas

garfenter commented on comments-ms/handlers.js:
```diff
@@ -0,0 +1,95 @@
+
+// Delete a COMMENT
+exports.deleteComment = function (request, reply) {
+    const user = {
```
La variable se podría llamar: "params"

garfenter commented on comments-ms/handlers.js:
```diff
@@ -0,0 +1,95 @@
+
+// Find all COMMENTs by dbid
+exports.findCommentByDbid = function (request, reply) {
+    request.app.db.query("SELECT * FROM PUBICATION WHERE state = 'ACTIVE' AND dbid = :dbid", (err, rows, fields) => {
```
El nombre de la tabla debería de ser: "COMMENT"

alexredle committed:
UPD: Se agrega paginacion en servicio de findAllComments

Se agrega paginacion en servicio de findAllComments
Se cambia nombre de recurso comment a comments
Se corrigen nombres en variables

alexredle committed:
UPD: Se actualiza valor default para limit 0,10 de findAll

garfenter commented on comments-ms/handlers.js:
```diff
@@ -0,0 +1,99 @@
+        limit: request.query.limit,
+        offset: request.query.offset
+    }    
+    request.app.db.query("SELECT * FROM COMMENT WHERE state = 'ACTIVE' LIMIT :limit, :offset", pagination,(err, rows, fields) => {
```
Nice!

garfenter commented on comments-ms/handlers.js:
```diff
@@ -0,0 +1,99 @@
+        dbid: request.params.dbid
+    };
+
+    request.app.db.query("UPDATE COMMENT SET state = 'DELETED' WHERE dbid = :dbid", params, (err, result) => {
```
@alexredle olvide mencionarte que habíamos acordado utilizar apostrofo como delimitador del string, ' en lugar de ", el problema entiendo que es por el "'DELETE'" en este caso lo que acordamos fue de que delete se pasaría como parametro dentro de la variable params.

garfenter commented on comments-ms/routes.js:
```diff
@@ -0,0 +1,60 @@
+            validate: {
+               query: {
+                    limit: Joi.number().integer().min(1).max(100).default(0),
+                    offset: Joi.number().integer().min(1).max(100).default(10)
```
Pienso que no deberíamos de limitar el offset en 100 ya que podríamos en dado caso paginar mas de 100 elementos.

alexredle committed:
UPD: Se actualiza querys con comilla simple

alexredle committed:
UPD: Se actualiza el offset de 10 a 100

alexredle committed:
FIX: Se corrige sintaxis de update.

alexredle committed:
FIX: tipo de dato

alexredle committed:
MOD: Se actualizan contexto de servicios

garfenter reviewed (approved):
LGTM
